Hamilton_Matthew/detail.html

```diff
@@ -0,0 +1,138 @@
+                <br>
+                <br>
+                <br>
+                % if product.__class__.__name__ == "BulkProduct":
+                    <input type="number" class="form-control quantity_input" placeholder = ${product.quantity | h}> &nbsp &nbsp
+                % endif
+                <button class = "btn btn-buy">Buy</button>
+                <br>
+                <br>
+            </div>      
+        </%block>
+
+<%block name = "content_right">   
+    <div class="col-md-2">
+      <h2 class = "myh2">Recently Viewed</h2>
+      <br/>
+
+      <ol class = "right_list">
+
+    % if int(request.session['new_prod']) in request.session['last_five'][1:6]:
+        Sup fam
+        <%
+            print("finding index, then prod_id")
+            index = request.session['last_five'].index(request.session['new_prod'])
+            print(index)
+
+            my_prod = request.session['last_five'][index]
+            print(my_prod)
+
+            print(request.session['last_five'])
+            request.session['last_five'].remove(my_prod)
+            print(request.session['last_five'])
+
+            request.session['last_five'].insert(0, my_prod)
+            print(request.session['last_five'])
+
+            request.last_five = []
+
+            for product_id in request.session['last_five']:
+                request.last_five.append(cmod.Product.objects.get(id = product_id))
+        %>
+    % endif
+
+
+    % for product in request.last_five[1:6]:
+      <a href="/catalog/detail/${product.id}">
+      <li>
+      <img src="${product.image_url() | h}" class = "right_pane_image" alt = "${product.name}" >
+      <br>
+      <br>
+      ${product.name | h}
+      <br>
+      <br>
+      </li>
+      </a>
+    % endfor
+      <br>
+      </ol>
+      
+        
+    </div>    
+
+
+
+
+
+<%doc>
+    % if int(request.session['new_prod']) in request.session['last_five'][1:6]:
+        Sup fam
+        <%
+            print("i'm inside the loop")
+            print(request.session['last_five'].index(int(request.session['new_prod'])))
+
+            index = 1
+            save_list = request.session['last_five']
+            print(save_list)
+
+            request.session['last_five'] = [request.session['new_prod']]
+            print(request.session['last_five'])
+
+            while index < save_list.index(int(request.session['new_prod'])):
+                request.session['last_five'].append(save_list[index])
+                index += 1
+            print(request.session['last_five'])
+
+            index += 1
+            while index < len(save_list):
+                request.session['last_five'].append(save_list[index])
+                index += 1
+                print(request.session['last_five'])
+
+            request.last_five = []
+
+            for product_id in request.session['last_five']:
+                request.last_five.append(cmod.Product.objects.get(id = product_id))
+        %>
+    % endif
+</%doc>
```

Comment:
Like previous comment, this could be removed by adding in detail.py: 

`if int(prod_id) in request.session['last_five']:
        # Requested to view Product that will show in right_list, need to remove
        request.session['last_five'].remove(int(prod_id))
        # Create request.last_five array to be rendered by right block in app_base.htm
        request.last_five = []
        for i in request.session['last_five']:
            request.last_five.append(cmod.Product.objects.get(id = i))`

Reply:
Done